src/workflows-app/components/ImportGithub.ts

```diff
@@ -123,7 +138,7 @@ const ImportGithub = ({ setLoading, signal, workspace, name, namespace, setSelec
                 },
                 false
               );
-              withBusyState(setLoading, submitMethod(signal, method, workspace, onSuccess, onError));
+              withBusyState(setLoading, () => submitMethod(signal, method, workspace, onSuccess, onError))();
```

Comment:
this may have masked a bug where setLoading may have not been being called properly since as this was, submitMethod was being called instantly and returning a promise, but withBusyState expects method that can be called later (deferred evocation).  Typescript alerted me to this because it was showing an error on expected withBusyState arg type.  This is one of many examples of what even a simple conversion to TS can surface.